onnx/onnx.in.proto

```diff
@@ -196,6 +204,83 @@ message NodeProto {
   optional string doc_string = 6;
 }
 
+// Training information
+// TrainingInfoProto defines a computation graph similar to the inference one
+// (i.e., ModelProto.graph) but represents a training algorithm. Each execution
+// of a TrainingInfoProto leads to the update of some initializers in its
+// TrainingInfoProto.algorithm.initializer and ModelProto.graph.initializer.
+//
+// If there are multiple TrainingInfoProto's, sequentially executing all of
+// them once consistutes a full training algorithm. The initializer update
+// happens at the end of each TrainingInfoProto's execution. If the targeted
+// training algorithm contains consecutive update stages (such as block
+// coordinate descent methods), the user needs to create a TrainingInfoProto
+// for each stage.
+//
+// The execution of TrainingInfoProto has two mode. User can choose to execute
+// either "algorithm" or "initialization". The computation specified in
+// "training" improves initializers based on the given training data. The
+// computation specified in "initialization" resets the model to its
+// initial state (as if no training data has been consumed).
+//
+// Sveral inference components are usable in TrainingInfoProto; the list
+// includes
+//   1. ModelProto.graph is a callble inside TrainingInfoProto.algorithm.
+//      Each call is a NodeProto in TrainingInfoProto.algorithm.node with
+//      NodeProto.op_type == "Graph" and
+//      NodeProto.name == ModelProto.graph.name.
+//   2. Initializers in ModelProto.graph.initializer can be referenced by
+//      nodes in TrainingInfoProto.algorithm.nodes and
+//      TrainingInfoProto.initialization.nodes. It implies that training
+//      nodes cannot produce values with names identical to inference
+//      initializers.
+message TrainingInfoProto {
+  // This field is a graph that describes a way to compute the initial tensors
+  // upon starting the training process. Initialization graph has no input
+  // and can have multiple outputs. Assigning those outputs to the associated
+  // initializers follows the relation defined by update_binding as if
+  // TrainingInfoProto.initialization is another training algorithm.
+  //
+  // By default, this field is an empty graph and its evaluation does not
+  // produce any output.
+  optional GraphProto initialization = 1;
+
+  // This fields is a training algorithm. Given required inputs, it computes
+  // outputs to update initializers in either its own initializer list or in
+  // ModelProto.graph.initializer. In general, this graph contains
+  // loss node, gradient node, optimizer node, increment of iteration count,
+  // and calls to the inference function.
+  //
+  // The field "algorithm.node" can reference functions defined in
+  // the field "ModelProto.function". The initializers defined in
```

Comment:
This sentence is no longer relevant and should be removed (same applies for the rest of the proto files).

Reply:
This paragraph is changed to
```
  // The field algorithm.node can invoke ModelProto.graph as a node.
  // The initializers defined in ModelProto.graph.initializer are visible in
  // the "algorithm"
```